Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -504,17 +728,86 @@ public ArraySegment<byte> GetPayloadBytes(FileAccessSetup setup, MemoryStream st
                 WriteInjectionTimeoutBlock(writer, timeoutMins);
                 WriteTranslationPathStrings(writer, DirectoryTranslator);
                 WriteErrorDumpLocation(writer, InternalDetoursErrorNotificationFile);
-                WriteFlagsBlock(writer);
+                WriteFlagsBlock(writer, m_fileAccessManifestFlag);
                 WriteExtraFlagsBlock(writer, FileAccessManifestExtraFlag.None);
                 WritePipId(writer, PipId);
                 WriteReportBlock(writer, setup);
                 WriteDllBlock(writer, setup);
-                WriteManifestTreeBlock(writer);
+                if (!IsManifestTreeBlockSealed)

```

Comment:
>IsManifestTreeBlockSealed [](start = 21, length = 25)

Very obfuscated that this is just a null-check for `m_sealedManifestTreeBlock` field. which MUST be read/written before. Using an indirection makes this code jump out as having a big bug, where it doesn't please juse use the `m_sealedManifestTreeBlock == null` check.

Reply:
in both serialize and deserialize

---
In reply to: [276814592](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276814592) [](ancestors = 276814592)